fhir/client.go

```diff
@@ -85,22 +101,23 @@ func (c *Client) NewSearchTypeRequest(resourceType string) (*http.Request, error
 // SearchTypeURL generates the URL for a search-type interaction of the given
 // resource type
 func (c *Client) SearchTypeURL(resourceType string) string {
-	return c.Base + "/" + resourceType
+	rel := &url.URL{Path: path.Join(c.baseURL.Path, resourceType)}
```

Comment:
This needs to be done since the subsequent call to `ResolveReference` overwrites any previously set path that is part of the base URL. Without this the request won't go through.

Reply:
AFAIK, the path is only replaced if the path you use in resolve starts with a slash. Please test if it works with the resourceType that doesn't contain any slashes.